### PR TITLE
Function to create basic/default yaml file

### PR DESCRIPTION
**Pull request by MerktSimon**

### CODE REVIEW

File: tests/test_yaml.py [hunk 1]
```diff
@@ -18,3 +18,7 @@ def test_validate():
     # should be well-formed
     file_ = "doc/example/example_Fujita/Fujita.yaml"
     validate(file_)
+
+
+def test_create_default_yaml():
```
Comment (dweindl) [inline tests/test_yaml.py]: Codacy Issue found: [Missing function or method docstring](https://app.codacy.com/manual/dweindl/PEtab/pullRequest?prid=5219106)

File: petab/yaml.py [hunk 2]
```diff
@@ -159,4 +162,20 @@ def write_yaml(yaml_config: Dict[str, Any], filename: str) -> None:
     """
 
     with open(filename, 'w') as outfile:
-        yaml.dump(yaml_config, outfile, default_flow_style=False)
+        yaml.dump(yaml_config, outfile, default_flow_style=False,
+                  sort_keys=False)
+
+
+def create_default_yaml(model_name):
```
Comment (dweindl) [inline petab/yaml.py]: Codacy Issue found: [Missing function or method docstring](https://app.codacy.com/manual/dweindl/PEtab/pullRequest?prid=5219106)

**Comment by yannikschaelte** [comment]
I agree that this function won't be too useful, since the employed default names are marked for removal already (they originated from back when there was no yaml yet but just standardized file names).

What would however make a lot of sense, not sure if it exists yet @dweindl , is the ability for a given petab problem (or rather a collection of file names) to write a yaml file linking all files.

**Comment by JanHasenauer** [comment]
I agree and vote for not include this

**Comment by dweindl** [comment]
> What would however make a lot of sense, not sure if it exists yet @dweindl , is the ability for a given petab problem (or rather a collection of file names) to write a yaml file linking all files.

Yeah, may make sense to extend `petab.Problem.to_files` by an optional `yaml_file` argument instead.

File: tests/test_yaml.py (hunk 1 above)
Comment (dweindl) [inline tests/test_yaml.py]: Codacy Issue found: [Missing function or method docstring](https://app.codacy.com/manual/dweindl/PEtab/pullRequest?prid=5219106)

File: petab/yaml.py (hunk 2 above)
Comment (dweindl) [inline petab/yaml.py]: Codacy Issue found: [Missing function or method docstring](https://app.codacy.com/manual/dweindl/PEtab/pullRequest?prid=5219106)

File: petab/yaml.py [hunk 3]
```diff
@@ -7,6 +7,9 @@
 import jsonschema
 import yaml
 from .C import *  # noqa: F403
+from . problem import get_default_condition_file_name, \
```
Comment (dweindl) [inline petab/yaml.py]: Codacy Issue found: [Unused get_default_sbml_file_name imported from problem](https://app.codacy.com/manual/dweindl/PEtab/pullRequest?prid=5219106)

File: tests/test_yaml.py [hunk 4]
```diff
@@ -18,3 +19,19 @@ def test_validate():
...
```
Comment (dweindl) [inline tests/test_yaml.py]: Codacy Issue found: [Missing function or method docstring](https://app.codacy.com/manual/dweindl/PEtab/pullRequest?prid=5219106)

**Comment by MerktSimon** [comment]
I got rid of all the `get_default_name...`functions. Now you have to enter file paths by hand.
It is also now included as optional in `problem.to_file()`. But I left it as a stand alone function so that users who have old petab folders can just use it to create yaml without detour over `petab.problem

File: tests/test_yaml.py (hunk 4 above)
Comment (dweindl) [inline tests/test_yaml.py]: Codacy Issue found: [Missing function or method docstring](https://app.codacy.com/manual/dweindl/PEtab/pullRequest?prid=5219106)

File: petab/yaml.py [hunk 5]
```diff
@@ -159,4 +158,36 @@ def write_yaml(yaml_config: Dict[str, Any], filename: str) -> None:
     """
 
     with open(filename, 'w') as outfile:
-        yaml.dump(yaml_config, outfile, default_flow_style=False)
+        yaml.dump(yaml_config, outfile, default_flow_style=False,
+                  sort_keys=False)
+
+
+def create_default_yaml(sbml_file: str,
```
Comment (dweindl) [inline petab/yaml.py]: Why 'default'? Maybe something like `{create|write}_problem_yaml`?

Directly allow for lists of filenames where they are allowed in yaml?

File: petab/yaml.py [hunk 6]
```diff
@@ -159,4 +158,36 @@ def write_yaml(yaml_config: Dict[str, Any], filename: str) -> None:
...
+    Arguments:
+        sbml_file: Path of SBML model file
+        condition_file: Path of condition file
+        measurement_file: Path of measurement file
+        parameter_file: Path of parameter file
+        observable_file: Path of observable file
+        yaml_file: Path to which YAML file should be written
+        visualization_file: Optional Path to visualization file
+    """
+    problem_dic = {"condition_files": [condition_file],
```
Comment (dweindl) [inline petab/yaml.py]: Replace string literals by corresponding constants in `petab.C`.

File: petab/yaml.py [hunk 7]
```diff
@@ -159,4 +158,36 @@ def write_yaml(yaml_config: Dict[str, Any], filename: str) -> None:
...
+                   "measurement_files": [measurement_file],
+                   "sbml_files": [sbml_file],
+                   "observable_files": [observable_file]}
+    if visualization_file is not None:
+        problem_dic.update({'visualization_files': [visualization_file]})
+    yaml_dic = {"parameter_file": parameter_file,
```
Comment (dweindl) [inline petab/yaml.py]: same here

File: tests/test_yaml.py [hunk 8]
```diff
@@ -18,3 +19,37 @@ def test_validate():
     # should be well-formed
     file_ = "doc/example/example_Fujita/Fujita.yaml"
     validate(file_)
+
+
+def test_create_problem_yaml():
```
Comment (dweindl) [inline tests/test_yaml.py]: Codacy Issue found: [Missing function or method docstring](https://app.codacy.com/manual/dweindl/PEtab/pullRequest?prid=5219106)

File: tests/test_yaml.py [hunk 9]
```diff
@@ -18,3 +19,37 @@ def test_validate():
...
+        yaml_file2 = tempfile.mkstemp(dir=folder)[1]
+
+        sbml_files = [sbml_file, sbml_file2]
+        condition_files = [condition_file, condition_file2]
+        measurement_files = [measurement_file, measurement_file2]
+        observable_files = [observable_file, observable_file2]
+        create_problem_yaml(sbml_files, condition_files, measurement_files,
+                            parameter_file, observable_files, yaml_file2)
+        validate(yaml_file2)
+
```
Comment (dweindl) [inline tests/test_yaml.py]: Codacy Issue found: [Trailing newlines](https://app.codacy.com/manual/dweindl/PEtab/pullRequest?prid=5219106)

File: tests/test_yaml.py (hunk 8 above)
Comment (dweindl) [inline tests/test_yaml.py]: Codacy Issue found: [Missing function or method docstring](https://app.codacy.com/manual/dweindl/PEtab/pullRequest?prid=5219106)

File: tests/test_yaml.py (hunk 9 above)
Comment (dweindl) [inline tests/test_yaml.py]: Codacy Issue found: [Trailing newlines](https://app.codacy.com/manual/dweindl/PEtab/pullRequest?prid=5219106)

File: tests/test_yaml.py (hunk 8 above)
Comment (MerktSimon) [reply tests/test_yaml.py]: this is not done for tests, right?

File: tests/test_yaml.py [hunk 10]
```diff
@@ -18,3 +19,36 @@ def test_validate():
     # should be well-formed
     file_ = "doc/example/example_Fujita/Fujita.yaml"
     validate(file_)
+
+
+def test_create_problem_yaml():
```
Comment (dweindl) [inline tests/test_yaml.py]: Codacy Issue found: [Missing function or method docstring](https://app.codacy.com/manual/dweindl/PEtab/pullRequest?prid=5219106)

Comment (MerktSimon) [reply tests/test_yaml.py]: this is not done for tests, right?

Comment (dweindl) [reply tests/test_yaml.py]: can be ignored, yeah

**Pull request merged**